druid-shell/src/platform/x11/window.rs

```diff
@@ -602,6 +602,42 @@ impl Window {
         }
     }
 
+    pub fn handle_wheel(&self, event: &xcb::ButtonPressEvent) -> Result<(), Error> {
+        let button = event.detail();
+        // The scroll delta of 120.0 was stolen from the GTK code, which apparently came from the
+        // windows docs.
+        let delta = match button {
```

Comment:
We should also check for whether the shift key is being held down and if so convert vertical scrolling into horizontal scrolling. This was introduced with #709 and you can copy the logic from the GTK code probably.